dist/tools/codespell/check.sh

```diff
@@ -56,8 +56,7 @@ if [ -n "${ERRORS}" ]
 then
     printf "%sThere are typos in the following files:%s\n\n" "${CERROR}" "${CRESET}"
     printf "%s\n" "${ERRORS}"
-    # TODO: return 1 when all typos are fixed
-    exit 0
+    exit 1
```

Comment:
I'm still very much afraid that this will cause more hassle with false positives than any wrong spelling ever would

Reply:
Same here. But #15138 showed that adding false positives to the list of ignored words is not so difficult. In a PR, codespell is also only run on modified files and this will also limit the unrelated typo found (because now we start at a point where all existing typos are fixed).

Reply:
I'm also a bit on the fence here, same argument as @benpicco. I feel this could be helped if maintainers start checking the output of travis more, even if its green and look at the suggestions.

Reply:
> this could be helped if maintainers start checking the output of travis more

You know that in practice this never happens (for many reasons)

Reply:
@aabadie pointed out offline that only two new typos went through in 20 days, so it doesn't seem to bad.